src/Hide.js

```diff
@@ -3,23 +3,26 @@ import PropTypes from 'prop-types'
 import Box from './Box'
 import theme from './theme'
 
-const getMaxWidth = em => em - 0.01
-
-const breakpoints = props => ({
-  xs: `@media screen and (max-width: ${getMaxWidth(
-    props.theme.breakpoints[0]
-  )}em)`,
-  sm: `@media screen and (min-width: ${
-    props.theme.breakpoints[0]
-  }em) and (max-width: ${getMaxWidth(props.theme.breakpoints[1])}em)`,
-  md: `@media screen and (min-width: ${
-    props.theme.breakpoints[1]
-  }em) and (max-width: ${getMaxWidth(props.theme.breakpoints[2])}em)`,
-  lg: `@media screen and (min-width: ${
-    props.theme.breakpoints[2]
-  }em) and (max-width: ${getMaxWidth(props.theme.breakpoints[3])}em)`,
-  xl: `@media screen and (min-width: ${props.theme.breakpoints[3]}em)`
-})
+const getMaxWidth = width => width - 0.01
+
+const breakpoints = props => {
+  const unit = props.theme.breakpointsUnit
+  return {
+    xs: `@media screen and (max-width: ${getMaxWidth(
+      props.theme.breakpoints[0]
+    )}${unit})`,
+    sm: `@media screen and (min-width: ${
+      props.theme.breakpoints[0]
+    }${unit}) and (max-width: ${getMaxWidth(props.theme.breakpoints[1])}${unit})`,
+    md: `@media screen and (min-width: ${
+      props.theme.breakpoints[1]
+    }${unit}) and (max-width: ${getMaxWidth(props.theme.breakpoints[2])}${unit})`,
+    lg: `@media screen and (min-width: ${
+      props.theme.breakpoints[2]
+    }${unit}) and (max-width: ${getMaxWidth(props.theme.breakpoints[3])}${unit})`,
+    xl: `@media screen and (min-width: ${props.theme.breakpoints[3]}${unit})`
+  }
```

Comment:
For some reason, I feel like adding breakpointUnit would add a bit of confusion. I would prefer having this following API:

if `breakpoints` is defined as an array on integers, e.g. [32, 40, 48, 64], it will be passed with the `em` unit.
if `breakpoints` is defined as an array pixels passed as strings, e.g. ['320px', '640px', '1024px'], it will be passed with the `px` unit. The same could be applied for %. 

Hence, getting the unit from the breakpoints array values instead as defined in the theme.

Reply:
interesting... I feel like we shouldn't have to set conditions to pick the unit based on what is in the array - string or number. 
the breakpoints array should be really kept as just integers imo. 
You can also see that `Hide` uses a util function that sets `maxWidth` by doing `width - 0.01`. I think this probably happens more that someone is doing some sort of math with the breakpoints.
If we force them to pass a string. Then in order for anyone to do any math, they would have to extract the number out of the string each time and then do what they need.